src/main/java/com/aws/greengrass/shadowmanager/sync/model/FullShadowSyncRequest.java

```diff
@@ -259,15 +265,21 @@ private void handleFirstLocalSync(@NonNull ShadowDocument cloudShadowDocument, l
      * Delete the local shadow using the request handlers and then update the sync information.
      *
      * @param syncInformation     The sync information for the thing's shadow.
-     * @param localShadowDocument The current local document.
      * @throws SkipSyncRequestException if the delete request encountered a skipable exception.
      */
-    private void handleLocalDelete(@NonNull ShadowDocument localShadowDocument,
-                                   @NonNull SyncInformation syncInformation) throws SkipSyncRequestException {
+    private void handleLocalDelete(@NonNull SyncInformation syncInformation) throws SkipSyncRequestException {
         deleteLocalShadowDocument();
+        // Since the local shadow has been deleted, we need get the deleted shadow version from the DAO.
+        long localShadowVersion = context.getDao().getDeletedShadowVersion(getThingName(), getShadowName())
+                .orElse(syncInformation.getLocalVersion() + 1);
         context.getDao().updateSyncInformation(SyncInformation.builder()
-                .localVersion(localShadowDocument.getVersion())
-                .cloudVersion(syncInformation.getCloudVersion())
+                .localVersion(localShadowVersion)
+                // If the cloud shadow was deleted, then the last synced version might be 1 higher than the last version
+                // that was synced.
+                // If the device was offline for a long time and the cloud shadow was deleted multiple times in that
```

Comment:
what does it mean by "was deleted multiple times"? what would the cloud version be in this case? does it keep incrementing?  
btw is there iot doc on how cloud deletion results in version increment?

Reply:
So if the cloud shadow was created and deleted multiple times, the version will keep incrementing for each update and each delete. So if the cloud shadow version was 10 and the device goes offline and the cloud shadow was updated 4 times and deleted 2 times, the version of the cloud shadow would now be 16.
Here is the [link](https://docs.aws.amazon.com/iot/latest/developerguide/device-shadow-data-flow.html#deleting-thing-data) to the documentation. There is a link in this section which talks about the version increment.